internal/provider/baremetal/inventory.go

```diff
@@ -13,10 +17,54 @@ func (p *baremetalProvider) CleanPlatformValuesFromDBUpdates(_ map[string]interf
 	return nil
 }
 
+// GetActualSchedulableMasters returns the value of the SchedulableMasters cluster
+// property in case it was actively set through the API. Otherwise it returns
+// the expected default value depending on the number of hosts in the cluster.
+func (p *baremetalProvider) GetActualSchedulableMasters(cluster *common.Cluster) (bool, error) {
+	if cluster == nil {
+		return false, errors.New("unexpected 'nil' cluster")
+	}
+	if cluster.SchedulableMasters != nil {
+		return *cluster.SchedulableMasters, nil
+	}
+	if common.CountActiveHosts(cluster) < 5 {
+		return true, nil
+	}
+	return false, nil
+}
+
 func (p *baremetalProvider) SetPlatformUsages(
 	_ *models.Platform,
 	usages map[string]models.Usage,
 	usageApi usage.API) error {
 	usageApi.Remove(usages, usage.PlatformSelectionUsage)
 	return nil
 }
+
+func (p baremetalProvider) GenerateManifests(cluster *common.Cluster) (map[string][]byte, error) {
```

Comment:
this func seems similar in vsphere in ovirt providers. can we extract it to a common location.

Reply:
done